app/services/doc_auth/error_generator.rb

```diff
@@ -362,12 +385,28 @@ def scan_for_unknown_alerts(response_info)
       unknown_fail_count
     end
 
-    def self.general_error(_liveness_enabled)
-      Errors::GENERAL_ERROR
+    # This method replicates TrueIdResponse::attention_with_barcode? and
+    # should be removed/updated when that is.
+    def attention_with_barcode_result(doc_auth_result, processed_alerts)
+      attention_result = doc_auth_result == LexisNexis::ResultCodes::ATTENTION.name
+      barcode_alerts = processed_alerts[:failed]&.count.to_i == 1 &&
+                       processed_alerts.dig(:failed, 0, :name) == '2D Barcode Read' &&
+                       processed_alerts.dig(:failed, 0, :result) == 'Attention'
+
+      attention_result && barcode_alerts ? true : false
```

Comment:
Is it equivalent to 

```ruby

doc_auth_result == LexisNexis::ResultCodes::ATTENTION.name && barcode_alerts

```

Reply:
Great point! Fixed in db9c081f6d7d2264c22c124cd0092424e39d6ac0. I separated out the name to a variable just to make it a little easier to read / reason about.